app/uk/gov/hmrc/gform/des/DesConnector.scala

```diff
@@ -31,17 +37,56 @@ import uk.gov.hmrc.http.logging.Authorization
 class DesConnector(wSHttp: WSHttp, baseUrl: String, desConfig: DesConnectorConfig) {
 
   def lookupRegistration(utr: String, desRegistrationRequest: DesRegistrationRequest)(
-    implicit ex: ExecutionContext): Future[DesRegistrationResponse] = {
+    implicit ex: ExecutionContext): Future[ServiceCallResponse[DesRegistrationResponse]] = {
+
     implicit val hc = HeaderCarrier(
       extraHeaders = Seq("Environment" -> desConfig.environment),
       authorization = Some(Authorization(s"Bearer ${desConfig.authorizationToken}")))
     Logger.info(s"Des registration, UTR: '$utr', ${loggingHelpers.cleanHeaderCarrierHeader(hc)}")
-    wSHttp.POST[DesRegistrationRequest, DesRegistrationResponse](
-      s"$baseUrl${desConfig.basePath}/registration/organisation/utr/$utr",
-      desRegistrationRequest)
 
+    wSHttp
+      .doPost[DesRegistrationRequest](
+        s"$baseUrl${desConfig.basePath}/registration/organisation/utr/$utr",
+        desRegistrationRequest,
+        List.empty[(String, String)]
+      )
+      .map { httpResponse =>
+        if (httpResponse.status === 400) {
+          processResponse[DesRegistrationResponseError, Nothing](httpResponse) { desError =>
+            if (desError.code === "INVALID_UTR") NotFound
+            else {
+              Logger.error("Problem when calling des registration: " + Json.prettyPrint(Json.toJson(desError)))
+              ServiceNotAvailable
```

Comment:
this is still within the scope of if (httpResponse.status === 400) 
I don't imagine responseStatus 400 representing service unavailable

Reply:
Please see https://github.com/hmrc/gform/pull/389/files/6d17606692afce5f72fbbc89d85f3c3683449b9a#r263312198